23.x backports

jonatack opened the pull request:
Backport the following to 23.x:

- #24468
- #24528
- #24527
- #24609
- #24555 
- #24663
- #24572 
- #24636 
- #24553 
- #24659
- #24521 
- #24523
- #24690

hebasto committed:
build: Fix Boost.Process test for Boost 1.78

Github-Pull: bitcoin#24523
Rebased-From: 532c64a

jonatack committed:
init, doc: improve -onlynet help and tor/i2p documentation

and harmonize them as follows

- s/outgoing/automatic outbound/
- s/Incoming/Inbound and manual/ (are not affected by this option.)
- s/only through network/only to network/
- s/this option. This option/this option. It/
- s/network types/networks/

and also pick up a few nits in doc/p2p-bad-ports.md

Github-Pull: bitcoin#24468
Rebased-From: a1db99a

jonatack committed:
doc: create initial doc/cjdns.md for cjdns how-to documentation

Adapted in part from the CJDNS description in #23077 by Vasil Dimov
and from CJDNS documentation and feedback by Caleb James DeLisle.

Co-authored-by: Vasil Dimov <vd@FreeBSD.org>

Github-Pull: bitcoin#24555
Rebased-From: ed15848

hebasto committed:
build: Fix Boost.Process detection on macOS arm64

Could be tested as follows:
```
% brew install boost@1.76
% ./autogen.sh
% ./configure --with-boost='/opt/homebrew/opt/boost@1.76'
```

Github-Pull: bitcoin#24521
Rebased-From: 1d4157a

mzumsande committed:
test: set segwit height back to 0 on regtest

This was changed in #22818 from 0 to 1. Since it changes
BLOCK_OPT_WIT of the genesis block, older versions of bitcoin
core would not read regtest directories created with newer versions
without a reindex.

Github-Pull: bitcoin#24527
Rebased-From: 5ce3057

jonatack committed:
rpc: rename getdeploymentinfo status-next to status_next

Github-Pull: bitcoin#24528
Rebased-From: 5d7c69b

fanquake commented:
These are missing the `Github-Pull` and `Rebased-From` metadata in the commit messages.

jonatack commented:
> These are missing the `Github-Pull` and `Rebased-From` metadata in the commit messages.

Thanks @fanquake, done.

theStack committed:
contrib: fix signet miner (sighash mismatch)

PSBT signing was changed to use SIGHASH_DEFAULT by default in #22514.
The signet miner script sets the sighash type of the created PSBT to
SIGHASH_ALL, hence this leads to a sighash mismatch when the
`walletprocesspsbt` RPC is called. Fix this by explicitly passing the
correct sighash type.

Note that the same change was needed in one of our functional tests,
see commit d3992669df826899a3de78a77a366dab46028026.

Reported by gruve-p.

Github-Pull: bitcoin#24553
Rebased-From: 12cc020

jonatack committed:
doc: update tor.md with cjdns and getnodeaddresses, fix tor grep,

and improve local addresses section

Github-Pull: bitcoin#24555
Rebased-From: 3bf6f0c

jonatack committed:
doc: update i2p.md with cjdns, improve local addresses section

Github-Pull: bitcoin#24555
Rebased-From: f44efc3

Commit:
ci: Temporarily use clang-13 to work around clang-14 TSan bug

Github-Pull: bitcoin#24572
Rebased-From: fa43933

jonatack committed:
Clarify in -maxtimeadjustment that only outbound peers influence time data

Github-Pull: bitcoin#24609
Rebased-From: 1bba72d

Commit:
rpc: Exclude descriptor when address is excluded

Github-Pull: bitcoin#24636
Rebased-From: faf37c2

fanquake committed:
util: add linkat to syscall sandbox (AllowFileSystem)

Github-Pull: bitcoin#24659
Rebased-From: 9809db3

jonatack committed:
doc, init: add links to doc/cjdns.md

Github-Pull: bitcoin#24663
Rebased-From: daae288

hebasto committed:
util: Add inotify_rm_watch to syscall sandbox (AllowFileSystem)

Github-Pull: bitcoin#24690
Rebased-From: f05a4cd

jonatack commented:
Updated along with the PR description with the merged pulls.

fanquake commented:
> with the merged pulls.

Are you going to add the rest of the merged pulls that need backporting?

jonatack commented:
Closing as these aren't considered as needing to be in v23.

MarcoFalke commented on src/rpc/blockchain.cpp:
```diff
@@ -1481,7 +1481,7 @@ static void SoftForkDescPushBack(const CBlockIndex* blockindex, UniValue& softfo
     bip9.pushKV("status", get_state_name(current_state));
     bip9.pushKV("since", g_versionbitscache.StateSinceHeight(blockindex->pprev, consensusParams, id));
-    bip9.pushKV("status-next", get_state_name(next_state));
+    bip9.pushKV("status_next", get_state_name(next_state));
```
If we don't backport this, it should be reverted in master again

jonatack replied:
IMO it should be backported along with the other pulls here, but it's not up to me.

MarcoFalke commented on src/chainparams.cpp:
```diff
@@ -397,7 +397,7 @@ class CRegTestParams : public CChainParams {
         consensus.BIP66Height = 1;  // Always active unless overridden
         consensus.CSVHeight = 1;    // Always active unless overridden
-        consensus.SegwitHeight = 1; // Always active unless overridden
+        consensus.SegwitHeight = 0; // Always active unless overridden
```
If we don't backport this, it should be reverted in master again

maflcko reviewed (commented):
Then they should be reverted in master

jonatack commented:
I think they should all be backported. But I don't decide.

jonatack commented:
Re-opened per IRC feedback.

hebasto commented:
> #24523 would be a good backport candidate too, if it happens to be merged before the next release candidate

It is ready for backporting now.

jonatack commented:
> > #24523 would be a good backport candidate too, if it happens to be merged before the next release candidate
> 
> It is ready for backporting now.

Done but there was a merge conflict in `configure.ac`. I took all the changes from that branch; don't hesitate to have a look.

fanquake commented on configure.ac:
```diff
@@ -1438,13 +1438,21 @@ if test "$use_external_signer" != "no"; then
+      dnl See: https://github.com/boostorg/process/issues/235
+      CXXFLAGS="$CXXFLAGS -Wno-error=narrowing"
+      TEMP_CPPFLAGS="$CPPFLAGS"
+      CPPFLAGS="$CPPFLAGS $BOOST_CPPFLAGS"
```
There are changes in 5dbed641eb118ab3f72a93f2b84dfc0220609f93 that are not part of the cherry-picked commit (532c64a7264dd3c7329e8839547837c57da7dbe8).

jonatack replied:
Done. Those were from the merge conflict. Updated with the following diff to make the commits the same:
```diff
diff --git a/configure.ac b/configure.ac
index a2f425ce93..a37553de8e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1458,8 +1458,6 @@ if test "$use_external_signer" != "no"; then
       dnl Boost 1.78 requires the following workaround.
       dnl See: https://github.com/boostorg/process/issues/235
       CXXFLAGS="$CXXFLAGS -Wno-error=narrowing"
-      TEMP_CPPFLAGS="$CPPFLAGS"
-      CPPFLAGS="$CPPFLAGS $BOOST_CPPFLAGS"
       TEMP_LDFLAGS="$LDFLAGS"
       dnl Boost 1.73 and older require the following workaround.
       LDFLAGS="$LDFLAGS $PTHREAD_CFLAGS"
@@ -1467,7 +1465,6 @@ if test "$use_external_signer" != "no"; then
         [have_boost_process="yes"],
         [have_boost_process="no"])
       LDFLAGS="$TEMP_LDFLAGS"
-      CPPFLAGS="$TEMP_CPPFLAGS"
       CXXFLAGS="$TEMP_CXXFLAGS"
       AC_MSG_RESULT([$have_boost_process])
       if test "$have_boost_process" = "yes"; then
```

hebasto replied:
The reason of the conflict is #24521 which has not been backported.

MarcoFalke replied:
Maybe it is easier to just backport @https://github.com/bitcoin/bitcoin/pull/24521 as well at this point?

jonatack replied:
> Maybe it is easier to just backport @#24521 as well at this point?

Done.

hebasto reviewed (approved):
~ACK c5cc0e5865c6ba4475d224a78361108f81e33c8d~ Sorry, retracted.

I've backported all commits locally, and got zero diff with this PR branch.

hebasto commented on doc/README.md:
```diff
@@ -73,6 +73,7 @@ The Bitcoin repo's [root README](/README.md) contains relevant information on th
 - [Assets Attribution](assets-attribution.md)
 - [Assumeutxo design](assumeutxo.md)
 - [bitcoin.conf Configuration File](bitcoin-conf.md)
+- [CJDNS Support](cjdns.md)
```
This link is dead.

jonatack replied:
Good catch. The commit adding that file is missing.

jonatack replied:
Thanks! Fixed the link at https://github.com/jonatack/bitcoin/blob/23.x-onlynet-help-and-doc-improvements/doc/README.md#miscellaneous by adding missing commits from #24555 and checked the other backports.

jonatack commented:
Addressed @hebasto's catch, added #24521 per https://github.com/bitcoin/bitcoin/pull/24512#discussion_r839261495, and updated the PR description.  #24710 looks RFM and would be good to add.

laanwj commented:
List-of-commits ACK 174af33d5d78cfe341b688f5393d9acc8c4aaf6b, I think we should merge this and move forward with rc3..

hebasto commented:
Why #24710 is listed among backported PRs?

hebasto reviewed (approved):
ACK 174af33d5d78cfe341b688f5393d9acc8c4aaf6b

I've backported all commits locally without any conflicts, and got zero diff with this PR branch.

> I think we should merge this and move forward with rc3.

Agree.

fanquake commented:
> I think we should merge this and move forward with rc3..

Yes. The list of commits has blown up, and this has dragged on long enough. I don't think there's a reason to wait for other documentation changes.

> Why https://github.com/bitcoin/bitcoin/pull/24710 is listed among backported PRs?

Not sure. I've removed it.

jonatack commented:
> Why #24710 is listed among backported PRs?

https://github.com/bitcoin/bitcoin/pull/24512#issuecomment-1084283299
